firestore/client.go

```diff
@@ -127,12 +159,25 @@ func (c *Client) path() string {
 }
 
 func withResourceHeader(ctx context.Context, resource string) context.Context {
-	md, _ := metadata.FromOutgoingContext(ctx)
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		md = metadata.New(nil)
+	}
 	md = md.Copy()
 	md[resourcePrefixHeader] = []string{resource}
 	return metadata.NewOutgoingContext(ctx, md)
 }
 
+func withRequestParamsHeader(ctx context.Context, requestParams string) context.Context {
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		md = metadata.New(nil)
+	}
+	md = md.Copy()
```

Comment:
I don't think this `Copy` call is necessary, `FromOutgoingContext` [builds the metadata from scratch](https://github.com/grpc/grpc-go/blob/v1.58.1/metadata/metadata.go#L260-L290), it's not pulling some reference directly from the context.